src/robotlibcore.py

```diff
@@ -101,36 +101,59 @@ def get_keyword_arguments(self, name):
         kw_method = self.__get_keyword(name)
         if kw_method is None:
             return None
-        args, defaults, varargs, kwargs = self.__get_arg_spec(kw_method)
-        if robot_version >= '3.2':
-            args += self.__new_default_spec(defaults)
-        else:
+        args, defaults, varargs, kwargs, kwonlydefaults = self.__get_arg_spec(kw_method)
+        if self.__rf_31:
             args += self.__old_default_spec(defaults)
+        else:
+            args += self.__new_default_spec(defaults)
         if varargs:
             args.append('*%s' % varargs)
         if kwargs:
             args.append('**%s' % kwargs)
+        if kwonlydefaults:
+            args += self.__kwonlydefaults_spec(kwonlydefaults)
         return args
 
+    @property
+    def __rf_31(self):
+        return robot_version < '3.2'
+
     def __new_default_spec(self, defaults):
         return [(name, value) for name, value in defaults]
 
     def __old_default_spec(self, defaults):
         return ['{}={}'.format(name, value) for name, value in defaults]
 
+    def __kwonlydefaults_spec(self, kwonlydefaults):
+        args = []
+        for argument, default_value in kwonlydefaults.items():
+            if self.__rf_31:
+                args.append(self.__old_kwonlydefaults_spec(argument, default_value))
+            else:
+                args.append(self.__new_kwonlydefaults_spec(argument, default_value))
+        return args
+
+    def __new_kwonlydefaults_spec(self, argument, default_value):
+        return (argument, default_value)
+
+    def __old_kwonlydefaults_spec(self, argument, default_value):
+        return '%s=%s' % (argument, default_value)
```

Comment:
This looks ok but it moving the RF version check to the helper would possibly simplify the code a bit:

```python
def __format_default(self, argument_name, default_value):
    if RF31:
        return '%s=%s' % (argument_name, default_value)
    return (argument_name, default_value)
```

After this change the `__kwonlydefaults_spec` could probably be turned to a list comprehension. As a nice bonus the same helper could be used also with normal defaults.